cli/src/commands/git.rs

```diff
@@ -784,13 +784,30 @@ fn cmd_git_push(
             }
         }
 
-        // TODO: Narrow search space to local target commits.
-        // TODO: Remove redundant CommitId -> Commit -> CommitId round trip.
-        let revision_commit_ids: HashSet<_> =
+        let use_default_revset =
+            args.branch.is_empty() && args.change.is_empty() && args.revisions.is_empty();
+        let revision_commit_ids = if use_default_revset {
+            let Some(wc_commit_id) = wc_commit_id else {
+                    return Err(user_error("Nothing checked out in this workspace"));
+                };
+            let current_branches_expression = RevsetExpression::remote_branches(
+                StringPattern::everything(),
+                StringPattern::Exact(remote.to_owned()),
+            )
+            .range(&RevsetExpression::commit(wc_commit_id))
+            .intersection(&RevsetExpression::branches(StringPattern::everything()));
+            let current_branches_revset = tx
+                .base_workspace_helper()
+                .evaluate_revset(current_branches_expression)?;
+            current_branches_revset.iter().collect()
+        } else {
+            // TODO: Narrow search space to local target commits.
+            // TODO: Remove redundant CommitId -> Commit -> CommitId round trip.
             resolve_multiple_nonempty_revsets(&args.revisions, tx.base_workspace_helper(), ui)?
                 .iter()
                 .map(|commit| commit.id().clone())
-                .collect();
+                .collect::<HashSet<_>>()
+        };
```

Comment:
nit: move type annotation to `let revision_commit_ids: ...` since both cases end with `.collect()`?